internal/oci/runtime_spoofed.go

```diff
@@ -0,0 +1,245 @@
+package oci
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/cri-o/cri-o/internal/log"
+	"github.com/cri-o/cri-o/pkg/config"
+	"github.com/cri-o/cri-o/utils/cmdrunner"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"k8s.io/client-go/tools/remotecommand"
+	types "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+// runtimeSpoofed is the Runtime interface implementation relying on conmon to
+// interact with the container runtime.
+type runtimeSpoofed struct {
+	*Runtime
+
+	root    string
+	handler *config.RuntimeHandler
+}
+
+// newRuntimeSpoofed creates a new runtimeSpoofed instance
+func newRuntimeSpoofed(r *Runtime, handler *config.RuntimeHandler) RuntimeImpl {
+	runRoot := config.DefaultRuntimeRoot
+	if handler.RuntimeRoot != "" {
+		runRoot = handler.RuntimeRoot
+	}
+
+	return &runtimeSpoofed{
+		Runtime: r,
+		root:    runRoot,
+		handler: handler,
+	}
+}
+
+
+// CreateContainer creates a container.
+func (r *runtimeSpoofed) CreateContainer(ctx context.Context, c *Container, cgroupParent string, restore bool) (retErr error) {
+
+	// Start sleep command to create process space for container
+	cmd := cmdrunner.Command("/bin/sleep","infinity") // nolint: gosec

```

Comment:
I would like to figure out a way to avoid having this process. IT doesn't looks like it's being stopped anywhere, and adds additional overhead. What was the motivation for having it?

Reply:
The original motivation was to create a process space that the CNI could still hook into as I started running into issues if the PID of the container was 0. However, I'm not sure if it's needed as we could just use the PID of crio.

If this duplicate process is actually needed I can add in process termination/cleanup into` StopContainer`.